File: Treino/Semana 1/quadrado_magico.cpp
#include <iostream>
#include <stdlib.h>

using namespace std;

int N;
int *sumColumns, *sumLines, sumMainDiagonal = 0, sumSecondaryDiagonal = 0;
int answer = -1;

int main()
{
    cin >> N;

    sumColumns = (int*)malloc(sizeof(int) * N);
    sumLines = (int*)malloc(sizeof(int) * N);

    for (int i = 0; i < N; i++)
    {
        sumColumns[i] = 0;
        sumLines[i] = 0;
    }

    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < N; j++)
        {
            int number;
            cin >> number;

            sumColumns[j] += number;
            sumLines[i] += number;

            if (i == j) sumMainDiagonal += number;
            if (j == N - 1 - i) sumSecondaryDiagonal += number;
        }
    }

    if (sumMainDiagonal == sumSecondaryDiagonal && sumMainDiagonal == sumColumns[0] && sumMainDiagonal == sumLines[0])
    {
        for (int i = 1; i < N; i++) 
        {
            if (sumColumns[i] != sumMainDiagonal || sumLines[i] != sumMainDiagonal) break;
            if (i + 1 == N) answer = sumMainDiagonal;
        }
    }
    cout << answer;
    return 0;
}

File: Treino/Semana 1/Daniel/quadrado_magico.py
# EXCEPTION
answer = -1

N = int(input())

sum_columns = [0 for i in range(N)]
sum_lines = [0 for i in range(N)]
sum_main_diagonal = 0
sum_secondary_diagonal = 0

for i in range(N):
    line = input().split(' ')
    for j in range(N):
        number = int(line[j])
        
        sum_columns[j] += number
        sum_lines[i] += number

        if i == j:
            sum_main_diagonal += number
        if j == N - 1 - i:
            sum_secondary_diagonal += number

if sum_main_diagonal == sum_secondary_diagonal and sum_main_diagonal == int(sum_columns[0]) and sum_main_diagonal == int(sum_lines[0]):
    for i in range(1, N):
        if sum_columns[i] != sum_main_diagonal or sum_lines[i] != sum_main_diagonal:
            break
        if i + 1 == N:
            answer = sum_main_diagonal
print(answer)